lib/fika/type_checker.ex

```diff
@@ -301,19 +301,13 @@ defmodule Fika.TypeChecker do
   end
 
   defp infer_if_else_blocks(env, if_block, else_block) do
-    {_, if_type_val, env} = infer_block(env, if_block)
-    {_, else_type_val, _} = type = infer_block(env, else_block)
-
-    unless if_type_val == else_type_val do
-      Logger.debug("if and else block have different types")
-
-      error =
-        "Expected if and else blocks to have same return type. " <>
-          "Got #{if_type_val} and #{else_type_val}"
-
-      {:error, error}
-    else
-      type
+    with {:ok, if_type_val, env} <- infer_block(env, if_block),
+         {:ok, else_type_val, _} <- infer_block(env, else_block) do
```

Comment:
We need to carry forward all "env"s that we get as a result of type inference. In this line we are ignoring it using _.

Reply:
nice, will fix this